src/services/completions.ts

```diff
@@ -2197,4 +2198,27 @@ namespace ts.Completions {
     function hasIndexSignature(type: Type): boolean {
         return !!type.getStringIndexType() || !!type.getNumberIndexType();
     }
+
+    function isInvalidTrigger(triggerCharacter: string | undefined, contextToken: Node, position: number): boolean {
+        switch (triggerCharacter) {
+            case '"':
+            case "'":
+            case "`":
+                switch (contextToken.kind) {
+                    case SyntaxKind.StringLiteral:
+                    case SyntaxKind.NoSubstitutionTemplateLiteral:
+                    case SyntaxKind.TemplateExpression:
+                    case SyntaxKind.TaggedTemplateExpression:
+                        break;
+                    default:
+                        Debug.fail();
+                }
+                return position === contextToken.parent.end;
+            case "<":
+                Debug.assert(contextToken.kind === SyntaxKind.LessThanToken);
+                return contextToken.parent.kind === SyntaxKind.BinaryExpression;

```

Comment:
There's technically no reason why `<` couldn't trigger type completions when the LHS is a type or has generic signatures.